scrapy/extensions/httpcache.py

```diff
@@ -453,14 +447,34 @@ def _parse_domain_from_url(self, spider, url):
     def _select_source(self, target, sources):
         return sources.keys()[0]
 
-    # Placeholder for now
     def _recompute_deltas(self, new_source, old_source, target_set):
-        #   For each key in the target set:
-        #       - read the delta _data and _length from the db
-        #       - decode the response with the old source
-        #       - encode the response with the new source
-        #       - write the new _data, _time, and _length to db
-        print('Chopper Dave says uh-oh, shouldn\'t hit this yet!\n')
+        for target_key in target_set:
+            # Get old response from the db
+            data = self._read_data(key_to_use=target_key, ignore_time=True)
+            # Decode serialized delta response with old source
+            old_serial_response = self._decode_response(data, old_source)
+            # Deserialze the 
+            data = self._deserialize(old_serial_response)
+            # Reconstruct original response
+            response = self._reconstruct_response(data)
+            # Serialize response so it can be encoded
+            target_response = self._serialize(response)
+            # Encode old response with new source
+            redelta = self._encode_response(target_response, new_source)
+            # Write new targets to db
+            batch = self._leveldb.WriteBatch()
+            batch.Put(target_key + b'_data', redelta)
+            batch.Put(target_key + b'_time', to_bytes(str(time())))
```

Comment:
Actually, let's just modify the data here and not the time, as the time stored should reflect the time we received a response from a server somewhere. If we constantly update the time when recomputing deltas, then it will impact the expiration feature's ability to work as intended.